File: 3301/3301.c
#include <stdio.h>

int main()
{
    int n;
    scanf("%d",&n);
    int money[8] = {50000,10000,5000,1000,500,100,50,10};
    int cnt = 0;
    int index = 0;
    while(n!=0)
    {
        if(n-money[index]>=0)
        {
            n -= money[index];
            cnt++;
        }else{
            index++;
        }
    }

    printf("%d",cnt);

    return 0;
}

File: 3301/3301.py
#3301

n = int(input())

money = [50000,10000,5000,1000,500,100,50,10]

cnt = 0
index = 0
while n!=0:
    if n - money[index] >= 0:
        n -= money[index]
        cnt += 1
    else : index += 1

print(cnt)
